### PR TITLE
Updates to Linux Pipeline

### PR DESCRIPTION
**Pull request by zachhannum**
This change contains a variety of updates for the Linux pipeline. 

I've added `deleteDir` as a post step to the agents that actually do the building. I think the last post step with `deleteDir` will only clean up the workspace on the top-level agent. 

I removed `reuseNode true` from the docker block since we are actually grabbing the linux agent for the matrix build and running within docker from the same step. 

I've also added `bionic` and `rhel8` to the `arm64` exclude list, which should fix the license-manager activation linker errors on the pro pipeline. The old pipeline didn't produce `bionic` and `rhel8` `arm64` builds.

### CODE REVIEW

File: jenkins/Jenkinsfile.linux [hunk 1]
```diff
@@ -63,7 +63,7 @@ def getAgentLabel(String arch) {
     return 'linux && arm64 && 4x'
```
Comment (MariaSemple) [inline jenkins/Jenkinsfile.linux:63]: Do you want to remove 4x here as well?

Comment (zachhannum) [reply jenkins/Jenkinsfile.linux]: Since Leticia just updated our x86 agents with more space, I've reverted this back to using the 4x agents instead!

**Review by AndrewMcClain — APPROVED** [review]
LGTM

**Review by MariaSemple — APPROVED** [review]
LGTM!

**Pull request merged**